### DISCUSSION TITLE
Using Morphia with missing classes

### DISCUSSION THREAD

**Discussion by MrGazdag** [MorphiaOrg/morphia#1979] [General]
**Please ask your question and provide whatever background or supporting information you think might be relevant:**
In a project of mine, i am using plugins to add additional data to specific database objects. Before I implement saving this data, i would like to know what happens if one of the plugins are not loaded. The plugins would store data in wrapper objects. Thus, the structure of the whole database object would look something like this:
```json
{
  ...,
  "pluginData": [
    {
      "key": "example_plugin:test_data",
      "data": {
        ...
      }
    }
  ]
}
```
The plugin here is called `example_plugin`, and the data key is called `test_data`.

Now, if the plugin called `example_plugin` is not loaded, then the data class will not be loaded. Will the query complete without errors? If it cannot deserialize the object at all (because of at least one missing class), then how should i go about adding plugin data to the database object? Do I need to add certain annotations to the `data` field?

**Please complete the following information:**
    Server Version: 5.0.8
    Driver Version: 4.2.3
    Morphia Version: 2.2.7

**Comment by evanchooly** [comment]
(apologies for the delay.  it seems my notifications are wonky...)  if those classes aren't present on the classpath *and* not represented in your entity, then it will likely just ignore that data depending on how your entities are structured.  There's also a possibility, e.g., if this data is stored in a `Map`, that that information, lacking any mapping information, would appear in the `Map` as `Document` instead.  I don't think there's any tests along these lines so the behavior is unclear.  I would suggest rigorous testing before turning it loose against production data since saving back any such "incomplete" entities would like end up dropping that data when the document is updated in the database and you'd lose data.
